mem/buffer_slice.go

```diff
@@ -0,0 +1,155 @@
+package mem
+
+import (
+	"io"
+)
+
+// BufferSlice offers a means to represent data that spans one or more Buffer
+// instances.
```

Comment:
It might be worth mentioning here that BufferSlice methods could panic if any of the underlying Buffer instances have been freed and there is an attempt to access data represented by that Buffer.